src/python/python/tests/test_util.py

```diff
@@ -285,3 +285,21 @@ def test05_render_fwd_assert(variants_all_ad_rgb):
     with pytest.raises(Exception) as e:
         img = mi.render(scene)
         dr.forward_to(img)
+
+
+def test06_variant_context():
+    mi.set_variant("scalar_rgb")
+
```

Comment:
```suggestion

if 'scalar_spectral' not in mi.variants():
    pytest.skip('Scalar spectral variant was not compiled.')
```

Reply:
Thanks! I'll combine both suggestions and make a robust variant detection with an (unlikely) skip if only `scalar_rgb` is built.